lib/src/rules/directives_ordering.dart

```diff
@@ -83,10 +124,24 @@ class _Visitor extends SimpleAstVisitor {
 
   @override
   void visitCompilationUnit(CompilationUnit node) {
+    _checkDartImportGoFirst(node);
+    _checkPackageImportBeforeRelative(node);
+  }
```

Comment:
I'm not a UX expert, but I'll share my intuition.

I think that in general, that kind of behavior tends to be confusing to users. Users, reasonably, tend to assume that if an error appears after they type that it had something to do with what they typed. Consider
```
class Car {}
Car c;
```
If a user types an `s` after `Car` on the last line, they will get an error and will rightly assume that the character they typed caused the error. But if a user moves an import from one place to another in order to fix an error, and then another error appears, they are likely to assume that (a) the change caused the second error and they need to find a different solution or (b) that the error didn't go away (if they don't notice that it really is a different error).

There is an exception if it's clear that the first error was masking the new errors (such as adding a type to a variable to fix one error that causes other errors in other locations to appear). I'm not sure whether the import case would be an exception, but I suspect that it wouldn't because I can't think of the logical dependency that they would have to assume.

But we might be able to get some concrete data to use to resolve the question by running the rule over some existing code. Data tends to be a good way to resolve questions.